cocos/2d/CCLabel.h

```diff
@@ -197,14 +197,58 @@ class CC_DLL Label : public Node, public LabelProtocol, public BlendProtocol
     * @param text The initial text.
     * @param hAlignment Text horizontal alignment.
     * @param maxLineWidth The max line width.
-    * @param imageOffset
     *
     * @return An automatically released Label object.
     * @see setBMFontFilePath setMaxLineWidth
     */
     static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
-        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0,
-        const Vec2& imageOffset = Vec2::ZERO);
+        const TextHAlignment& hAlignment = TextHAlignment::LEFT, int maxLineWidth = 0);
+
+    /**
+    * Allocates and initializes a Label, with a bitmap font file.
+    *
+    * @param bmfontPath A bitmap font file, it's a FNT format.
+    * @param text The initial text.
+    * @param hAlignment Text horizontal alignment.
+    * @param maxLineWidth The max line width.
+    * @param imageRect
+    * @param imageRotated
+    *
+    * @return An automatically released Label object.
+    * @see setBMFontFilePath setMaxLineWidth
+    */
+    static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
+        const TextHAlignment& hAlignment, int maxLineWidth, const Rect& imageRect, bool imageRotated);
+
+    /**
+    * Allocates and initializes a Label, with a bitmap font file.
+    *
+    * @param bmfontPath A bitmap font file, it's a FNT format.
+    * @param text The initial text.
+    * @param hAlignment Text horizontal alignment.
+    * @param maxLineWidth The max line width.
+    * @param subTextureKey Name of entry in PLIST texture atlas/sprite sheet
+    *
+    * @return An automatically released Label object.
+    * @see setBMFontFilePath setMaxLineWidth
+    */
+    static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
+        const TextHAlignment& hAlignment, int maxLineWidth, const std::string& subTextureKey);
```

Comment:
Can subTextureKey and imageRotated used together?

Reply:
No, they can't, and they shouldn't be, since the method to accept a `subTextureKey` is purely for convenience.   The `subTextureKey` is used to load the SpriteFrame, which internally has the data on whether the texture is rotated or not.

I was considering an alternative implementation, which would accept the `SpriteFrame*` instead, like this:

```
static Label* createWithBMFont(const std::string& bmfontPath, const std::string& text,
        const TextHAlignment& hAlignment, int maxLineWidth, SpriteFrame* frame);
```

but that would have resulted in more typing:
```
auto frame = spriteCache->getSpriteFrameByName(subTextureKey);
auto label = Label::createWithBMFont("fontname.fnt", "text", TextHAlignment::CENTER, 0, frame);
```

That was my rational for adding the convenience method that accepts the `subTextureKey`.